packages/common/src/utils/mapObject.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Map each key of a source object via a given valueMap function
+ */
+export function mapObject<
+  Source extends Record<string | number | symbol, unknown>,
+  Target extends { [key in keyof Source]: unknown }
+>(source: Source, valueMap: (key: keyof Source, value: Source[typeof key]) => Target[typeof key]): Target {
+  const target: Record<string, unknown> = {};
+  for (const [key, value] of Object.entries(source)) {
+    target[key] = valueMap(key, value as Source[keyof Source]);
+  }
+  return target as Target;
```

Comment:
totally not necessary here and no idea if there are perf consequences but my most used pattern for this is `Object.fromEntries(Object.entries(...).map(...))` to keep it functionally pure/immutable

```ts 
return Object.fromEntries(Object.entries(source).map(([key, value]) => [key, valueMap(key, value)]))l
```

Reply:
I like it! updated